tools/data-handler/src/commands/calculate.ts

```
@@ -631,8 +624,8 @@ export class Calculate {
         `Graph: Failed to read image file after generating graph: ${e}`,
       );
     } finally {
-      if (pathExists(filePath)) await rm(filePath);
-      if (pathExists(filePath + '.png')) await rm(filePath + '.png');
+      await rm(filePath, { force: true });
+      await rm(filePath + '.png', { force: true });
```

This is a bit risky, since clingraph sometimes produces the dot file(no extension) and it should always generate the png file. Double try catch is kind of ugly too. I think it will work more likely if the png is deleted first. This isn't the best fix, but clingraph will be replaced soon I think it is fine

---

I don't see the risk; unless the risk was already in the original code.
See: https://nodejs.org/api/fs.html#fspromisesrmpath-options
"force [<boolean>]When true, exceptions will be ignored if path does not exist."

---

My bad, did not know force skips exceptions. And yes the risk was in the code already, but it would happen if someone managed to delete the file between `pathExists` and `rm`, which would happen pretty much never. But now that rm does not throw, that issue cannot occur